<?xml version='1.0'?>
<sdf version="1.6">
  <world name="empty">

    <physics name="1ms" type="ignored">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>10.0</real_time_factor>
    </physics>

    <plugin filename="gz-sim-physics-system" name="gz::sim::systems::Physics">
    </plugin>
    <plugin filename="gz-sim-sensors-system" name="gz::sim::systems::Sensors">
      <render_engine>ogre2</render_engine>
    </plugin>
    <plugin filename="gz-sim-user-commands-system" name="gz::sim::systems::UserCommands">
    </plugin>
    <plugin filename="gz-sim-scene-broadcaster-system" name="gz::sim::systems::SceneBroadcaster">
    </plugin>

    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>1 1 1 1</diffuse>
      <specular>0.5 0.5 0.5 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name="base_plate">
      <pose>90 20 0 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>180 40</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.6 0.6 0.6 1</ambient>
            <diffuse>0.6 0.6 0.6 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>

    <model name="ground_plane">
      <pose>90 20 -0.1 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>180 40</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.2 0.2 0.2 1</ambient>
            <diffuse>0.2 0.2 0.2 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>


    <model name="track_01">
      <pose>0 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_01.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_01.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    
    <model name="track_02">
      <pose>30 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_02.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_02.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    
    <model name="track_03">
      <pose>60 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_03.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_03.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    
    <model name="track_04">
      <pose>90 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_04.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_04.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="track_05">
      <pose>120 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_05.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_05.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    
    <model name="track_06">
      <pose>150 0 0 0 0 0</pose>
      <static>true</static>
      <link name="track">
        <visual name="track_vis">
          <geometry>
            <mesh>
              <uri>../meshes/track_06.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
        <collision name="track_col">
          <geometry>
            <mesh>
              <uri>../meshes/track_06.stl</uri>
              <scale>0.001 0.001 0.001</scale>
            </mesh>
          </geometry>
        </collision>
      </link>
    </model>
    
  </world>
</sdf>